modules/javafx.controls/src/main/java/javafx/scene/control/skin/ListViewSkin.java

```diff
@@ -678,4 +678,16 @@ private int onScrollPageUp(boolean isFocusDriven) {
         flow.scrollTo(firstVisibleCell);
         return newSelectionIndex;
     }
+
+    private void horizontalUnitScroll(boolean right) {
+        if (getSkinnable().getEffectiveNodeOrientation() == NodeOrientation.RIGHT_TO_LEFT) {
+            right = !right;
+        }
+        ScrollBar sb = flow.getHbar();
```

Comment:
The ListView has the orientation property ``ListView.setOrientation(Orientation)``.
I think it should be
`````java
ScrollBar sb = flow.isVertical() ? flow.getHbar() : flow.getVbar();
`````
and the name of the method something like ``crossUnitScroll(boolean)``.

Reply:
I am afraid this suggestion is incorrect: the new key bindings affect horizontal scroll bar regardless of the orientation.

The new behavior can be checked with the Monkey Tester
https://github.com/andy-goryachev-oracle/MonkeyTest